### PR TITLE
Document Clearance testing helpers

### PR DESCRIPTION
**Pull request by derekprior**

### CODE REVIEW

**Comment by seanpdoyle** [comment]
Could we provide a thorough commit message documenting the changes?

**Comment by seanpdoyle** [comment]
Do these comments hook into HTML documentation generation?

**Comment by mike-burns** [comment]
These are beautiful.

@seanpdoyle this is written using YARD syntax, which rubydoc.info makes use of.

File: lib/clearance/testing/deny_access_matcher.rb [hunk 1]
```diff
@@ -1,10 +1,40 @@
...
   module Testing
+    # Provides matchers to be used in your controller specs.
+    # These are typically exposed to your controller specs by
+    # requiring `clearance/rspec` or `clearance/test_unit` as
+    # appropriate in your `rails_helper.rb` or `test_helper.rb`
+    # files.
     module Matchers
+      # The `deny_access` matcher is used to assert that a
+      #   request is denied access by clearance.
+      # @option opts [String] :flash The expected flash notice message. Defaults
```
Comment (seanpdoyle) [inline lib/clearance/testing/deny_access_matcher.rb]: Should we provide an example of what it would look like to use one of these options?

Comment (derekprior) [inline lib/clearance/testing/deny_access_matcher.rb]: I thought about that, but documenting aything beyond a contrived example would require a bit more code (including a sample clearance configuration) for a realistic example. In the end, I think the options are documented well enough for now.

**Pull request merged**